eth/downloader/downloader.go

```diff
@@ -1380,6 +1390,13 @@ func (d *Downloader) processHeaders(origin uint64, td, ttd *big.Int, beaconMode
 							rollback = 1
 						}
 					}
+					if len(rejected) != 0 {
```

Comment:
if the last chunkHeader is the final PoW-block, then `len(rejected)` will be `0`, but we still would have wanted to enter here, no?